autogluon_zeroshot/metrics/_roc_auc_cpp/__init__.py

```diff
@@ -29,10 +31,44 @@ def roc_auc_score(self, y_true: np.array, y_score: np.array) -> float:
         Args:
             y_true (np.array): 1D numpy array of dtype=np.bool8 as true labels.
             y_score (np.array): 1D numpy array of dtype=np.float32 as probability predictions.
-            sample_weight (np.array): 1D numpy array as sample weights, optional.
         Returns:
             float: AUC score
         """
         n = len(y_true)
         result = self._handle.cpp_auc_ext(y_score.astype(np.float32), y_true, n)
         return result
+
+    def _compile(self):
```

Comment:
I think one potential issue is that this is not an atomic operation in multithreading. Currently I am only initializing this class in ray threads, so multiple ray workers could try to compile at the same time. A work-around is to avoid the lazy import (which I was doing to avoid needing to compile) and just import automatically prior to using ray.

Reply:
Can you move the lazy import in `EnsembleSelectionConfigScorer` to the top of file so the compilation happens prior to using ray? I think that should avoid any issues

Reply:
Sure, moved the file there. The compilation should happen at the import which happens before ray is initialized so it should not cause any problem.